src/libcrun/handlers/wasmedge.c

```diff
@@ -34,6 +34,11 @@
 #  include <dlfcn.h>
 #endif
 
+#ifdef HAVE_WASMEDGE_LEGACY
+// user probably wanted to set both, so set it for them
+#  define HAVE_WASMEDGE 1
+#endif
+
```

Comment:
is there any value in supporting these older releases?

Reply:
@flouthoc you asked for it, why should we care about older releases?

Reply:
One issue i think is that updating crun version silently breaks wasm support since in most of the distros this is not a dependency of crun.

Reply:
My intention was to make sure in next update of crun existing wasmedge infra does not breaks. I think if we don't decide to support this should we error out that `following version of wasmedge` is not supported.

Reply:
If we can drop the older wasmedge support, I will suggest that to remove all `wasmedge_process` and `plugin` system parts currently. The reasons are below:

1. `WasmEdge process` is an experimental extension, which is not in the spec of WASM and WASI. Also, I would like to disable it until it's ready to be used in the production environment (e.g. crun). That's why the WasmEdge team decides to split from the standard `libwasmedge.so`.
2. The plugin system is also in active development, and there will be more unstable facts in these components. I would like to add it back once the features and functionalities are ready. Maybe 0.11 or later versions.

My apologies for enabling this component in the previous PR to make this error `crun: could not get wasmedge process module context` occur.

Reply:
@hown3d If you can drop support for `HAVE_WASMEDGE_LEGACY` from this PR we can merge this one and I'll open a new PR for https://github.com/containers/crun/pull/933#discussion_r904997340

Reply:
> All good, I can wait for the followup PR with this one. Currently not in a rush.

I'll open a new PR with commits from this included :)

Reply:
@hown3d Working on this?